corehq/ex-submodules/pillowtop/index_settings.py

```diff
@@ -1,25 +1,59 @@
+from copy import copy
+
+from django.conf import settings
+
 from corehq.elastic import SIZE_LIMIT
 
-INDEX_REINDEX_SETTINGS = {
-    "index": {
-        "refresh_interval": "1800s",
-        # this property will be filtered out on ES 1
-        "max_result_window": SIZE_LIMIT,
-        # todo: remove these deprecated properties we are off ES 1
-        "merge.policy.merge_factor": 20,
-        "store.throttle.max_bytes_per_sec": "1mb",
-        "store.throttle.type": "merge",
-    }
-}
-INDEX_STANDARD_SETTINGS = {
-    "index": {
-        "refresh_interval": "5s",
-        # this property will be filtered out on ES 1
-        "max_result_window": SIZE_LIMIT,
-        # todo: remove these deprecated properties we are off ES 1
-        "merge.policy.merge_factor": 10,
-        "store.throttle.max_bytes_per_sec": "5mb",
-        "store.throttle.type": "node",
-
-    }
+disallowed_settings_by_es_version = {
+    1: ['max_result_window'],
+    2: [
+        'merge.policy.merge_factor',
+        'store.throttle.max_bytes_per_sec',
+        'store.throttle.type'
+    ],
+    7: [
+        'merge.policy.merge_factor',
+        'store.throttle.max_bytes_per_sec',
+        'store.throttle.type'
+    ]
 }
+
+def _get_es_settings(es_settings):
+    es_settings = copy(es_settings)
```

Comment:
`copy` is not necessary since all of the functions that call this always pass in a new/unshared dict instance.
```suggestion
```

Reply:
Sure